### PR TITLE
:warning: (go/v3) deprecate go/v3 in favor of go/v4

### PR DESCRIPTION
**Pull request by camilamacedo86**
## Description 

deprecated go/v3 plugin

Motivation: https://github.com/kubernetes-sigs/kubebuilder/issues/3117

### CODE REVIEW

File: pkg/plugins/golang/v4/plugin.go [hunk 1]
```diff
@@ -63,3 +63,10 @@ func (p Plugin) GetCreateWebhookSubcommand() plugin.CreateWebhookSubcommand {
 
 // GetEditSubcommand will return the subcommand which is responsible for editing the scaffold of the project
 func (p Plugin) GetEditSubcommand() plugin.EditSubcommand { return &p.editSubcommand }
+
+func (p Plugin) DeprecationWarning() string {
+	return "This version is deprecated." +
+		"The `go/v3` cannot scaffold projects using kustomize versions v4x+ and cannot be fulled support on Kubernetes 1.25+." +
```
Comment (camilamacedo86) [inline pkg/plugins/golang/v4/plugin.go:69]:
```suggestion
		"The `go/v3` cannot scaffold projects using kustomize versions v4x+" +
		" and cannot be whole support on Kubernetes 1.25+." +
```

File: docs/book/src/plugins/go-v3-plugin.md [hunk 2]
```diff
@@ -1,4 +1,16 @@
...
+The `go/v3` cannot fully support Kubernetes 1.25+ and work with Kustomize versions > v3.
+
+The recommended way to migrate a `v3` project is to create a new `v4` project and copy over the API
+and the reconciliation code. The conversion will end up with a project that looks like a native `v4` project.
+For further information check the [Migration guide](../migration/migration_guide_gov3_to_gov4.md)
+
+</aside>
+
 
 Kubebuilder tool will scaffold the go/v3 plugin by default. This plugin is a composition of the plugins ` kustomize.common.kubebuilder.io/v1` and `base.go.kubebuilder.io/v3`. By using you can scaffold the default project which is a helper to construct sets of [controllers][controller-runtime].
```
Comment (everettraven) [inline docs/book/src/plugins/go-v3-plugin.md:15]: Is this plugin still the default until the go/v4 plugin is stabilized?

File: pkg/plugins/golang/v4/plugin.go [hunk 3]
```diff
@@ -63,3 +63,11 @@ func (p Plugin) GetCreateWebhookSubcommand() plugin.CreateWebhookSubcommand {
 
 // GetEditSubcommand will return the subcommand which is responsible for editing the scaffold of the project
 func (p Plugin) GetEditSubcommand() plugin.EditSubcommand { return &p.editSubcommand }
+
+func (p Plugin) DeprecationWarning() string {
+	return "This version is deprecated." +
+		"The `go/v3` cannot scaffold projects using kustomize versions v4x+" +
+		" and cannot be whole support on Kubernetes 1.25+." +
```
Comment (everettraven) [inline pkg/plugins/golang/v4/plugin.go:70]: Nit:
```suggestion
		" and cannot fully support Kubernetes 1.25+." +
```

**Review by everettraven — APPROVED** [review]
Overall looks good to me. Just a couple comments:

File: docs/book/src/plugins/go-v3-plugin.md (hunk 2 above)
Comment (camilamacedo86) [reply docs/book/src/plugins/go-v3-plugin.md:15]: Yes, it changes in another PR, see: https://github.com/kubernetes-sigs/kubebuilder/pull/3237 

Here we are only deprecating.

**Pull request merged**